packages/mobile/src/escrow/saga.ts

```diff
@@ -278,19 +281,28 @@ function* withdrawFromEscrowUsingPepper(komenciActive: boolean = false) {
       phoneHash
     )
 
-    const msgHash = contractKit.web3.utils.soliditySha3({ type: 'address', value: walletAddress })
+    const paymentIdSet: Set<string> = new Set()
+    for (const paymentId of escrowPaymentIds) {
+      paymentIdSet.add(paymentId)
+    }
+
     const context = newTransactionContext(TAG, 'Withdraw from escrow')
     // TODO: Batch the tranasctions and submit them together via `executeTransactions`
     // method on an instance of the MTW then submitting like usual
     const withdrawTxSuccess: boolean[] = []
-    for (let i = 0; i < escrowPaymentIds.length; i += 1) {
+    // Using an upper bound of 1000 to be sure this doesn't run forever
+    for (let i = 0; i < 1000; i += 1) {
+      const { paymentId, privateKey } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
+      if (!paymentIdSet.has(paymentId)) {
+        continue
+      }
+
```

Comment:
A quick suggestion to avoid looping 1000 times in the common case that all payments are processed.
```suggestion
    for (let i = 0; i < 1000 && paymentIdSet.size > 0; i += 1) {
      const { paymentId, privateKey } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
      if (!paymentIdSet.has(paymentId)) {
        continue
      }
      paymentIdSet.delete(paymentId)

```